source/extensions/filters/udp/dns_filter/dns_filter_utils.cc

```diff
@@ -89,6 +89,19 @@ absl::string_view getDomainSuffix(const absl::string_view name) {
   return name.substr(pos + 1);
 }
 
+absl::string_view getVirtualDomainName(const absl::string_view domain_name) {
+  // We can use names started with '.' as wildcard records in virtual domain name config
+  // since these are not valid domain names and in this way we optimize for future search against
+  // them. We expect only names like *.foo.com as a valid wildcard records because any other
+  // wildcard usages are not considered as valid ones, i.e. **.foo.com, *foo.bar.com, foo*.bar.com
+  // are all invalid.
+  if (domain_name.starts_with("*.")) {
+    return domain_name.substr(1);
```

Comment:
What if the domain_name is invalid, like just "*.".  This will return an empty string?

Reply:
If domain is `*.` it will return just `.`. You can see that in tests.
While working on these code I made few assumptions and cut the corners sometimes. Mostly because current code does not have any validation for passed domain names and in proto validations are: length >= 1 + header value regex, that's probably incorrect.
I did not want to increase the scope of my changes so decided not to cover validations. It is probably should be addressed, likely in different PR tho